### PR TITLE
Implement BFT Message receiver

### PR DESCRIPTION
**Pull request by liam-lai**
Implement BFT Message receiver. bfter to handler vote / syncinfo / timeout message.
additional feature: cache on received message, avoid spawn on the resource.

### CODE REVIEW

**Review by wgr523 — APPROVED** [review]
Can we rename `TC` to `TimeoutCert`?
(`QC` also to `QuorumCert`)

File: eth/bft/bft.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		broadcast:  broadcasts,
+	}
+}
+
+func (b *BFT) Vote(vote interface{}) {
+	b.engine.collectVote(vote)
+}
+
+func (b *BFT) Timeout(timeout interface{}) {
+	b.engine.collectTimeout(timeout)
```
Comment (wjrjerome) [inline eth/bft/bft.go]: collect TC? I'm getting confused with `timeout` which is self timeout in the machine VS `TC` which is a msg send through from different peers

File: eth/bft/bft.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+func New(engine *XDPoS.XDPoS, broadcasts BroadcastFns) *BFT {
+	consensus := ConsensusFns{
+		collectVote:    engine.CollectVote,
+		collectTimeout: engine.CollectTimeout,
+		updateRound:    engine.UpdateRound,
+	}
+	return &BFT{
+		messageBus: make(chan interface{}),
```
Comment (wjrjerome) [inline eth/bft/bft.go]: Is this message bus shared for all the below:
- BFT channel
- Timer channel
Or if this bus is only for the BFT channel usage?

Comment (liam-lai) [reply eth/bft/bft.go]: Only for BFT messages.

File: eth/bft/bft.go (hunk 1 above)
Comment (liam-lai) [reply eth/bft/bft.go]: Timeout
TimeoutCert

**Pull request merged**